### PR TITLE
[AV2-841] Feature/av2 841/update program switcher component to experience switcher

### PR DESCRIPTION
**Pull request by sasangachathumal**
**Story/Ticket Reference ID from JIRA:**
https://intersective.atlassian.net/browse/AV2-841

### Additional Comments

**This PR only have the code changes. so unit test in PR is falling. And I have create separate PR and branch for Unit testing because changes getting bigger so it's hard to code review track.** 

this is the other PR I created for Unit testing and i'm still working on it. https://github.com/intersective/app-v2/pull/1240

### CODE REVIEW

**Comment by sasangachathumal** [comment]
This PR only have the code changes. so unit test in PR is falling. And I have create separate PR and branch for Unit testing because changes getting bigger so it's hard to code review track.

Can you guys review this PR for me.

this is the other PR I created for Unit testing and i'm still working on it. https://github.com/intersective/app-v2/pull/1240

File: src/app/switcher/switcher.service.ts [hunk 1]
```diff
@@ -129,6 +188,9 @@ export class SwitcherService {
   }
 
   switchProgram(programObj: ProgramObj): Observable<any> {
+    // initialise Pusher and apollo here if there stack info in storage
+    this.sharedService.initPusherApollo();
```
Comment (trtshen) [inline src/app/switcher/switcher.service.ts]: Hi @sasangachathumal I prefer to call this "initWebServices" so we can reuse that for different services in the future.

File: src/app/switcher/switcher.service.ts [hunk 2]
```diff
@@ -76,23 +80,78 @@ export class SwitcherService {
...
     let imagewidth = 600;
-    programs.forEach(program => {
-      if (program.project.lead_image) {
-        const imageId = program.project.lead_image.split('/').pop();
-        if (!this.utils.isMobile()) {
-          imagewidth = 1024;
+    const expreancesList = [];
+    apiResults.forEach(result => {
+      const data = result.data;
+      data.Timelines.map(
```
Comment (trtshen) [inline src/app/switcher/switcher.service.ts]: @sasangachathumal I'm not sure if this possible, do we need to cover condition where the  `Timelines` object from data is null or undefined?

File: src/app/switcher/switcher-program/switcher-program.component.ts [hunk 3]
```diff
@@ -42,50 +43,36 @@ export class SwitcherProgramComponent extends RouterEnter implements AfterConten
...
   onEnter() {
     this.newRelic.setPageViewName('program switcher');
-    this.switcherService.getPrograms()
-      .subscribe(programs => {
-        this.programs = programs;
-        this._getProgresses(programs);
+    this.switcherService.getExperience(this.stacks).subscribe(
+      experiences => {
+        this.isExperiencesLoading = false;
+        this.programs = experiences;
```
Comment (shawnm0705) [inline src/app/switcher/switcher-program/switcher-program.component.ts]: let's name it consistently among the codebase, either "program" everywhere, or "experience" everywhere.

File: src/app/switcher/switcher.service.ts (hunk 1 above)
Comment (sasangachathumal) [reply src/app/switcher/switcher.service.ts]: sure. will change the naming.

File: src/app/switcher/switcher.service.ts (hunk 2 above)
Comment (sasangachathumal) [reply src/app/switcher/switcher.service.ts]: i will check, better to have something like that. i will check and add.

File: src/app/switcher/switcher.service.ts [hunk 4]
```diff
@@ -76,23 +80,78 @@ export class SwitcherService {
...
+        timeline => {
+          if (!this.utils.has(timeline, 'Program.config.theme_color')) {
+            if (!this.utils.has(timeline.Program, 'config')) {
+              timeline.Program.config = {
+                theme_color: 'var(--ion-color-primary)'
+              };
+            } else {
+              timeline.Program.config.theme_color = 'var(--ion-color-primary)';
```
Comment (shawnm0705) [inline src/app/switcher/switcher.service.ts]: I think @trtshen has latest code changes related to the color. @sasangachathumal you may want to implement them here as well

File: src/app/switcher/switcher.service.ts [hunk 5]
```diff
@@ -76,23 +80,78 @@ export class SwitcherService {
...
+            }
+          }
+          if (timeline.Project.lead_image) {
+            const imageId = timeline.Project.lead_image.split('/').pop();
+            if (!this.utils.isMobile()) {
+              imagewidth = 1024;
+            }
+            timeline.Project.lead_image = `${cdn}${imagewidth}/${imageId}`;
+          }
+          expreancesList.push({
```
Comment (shawnm0705) [inline src/app/switcher/switcher.service.ts]: typo

**Review by shawnm0705 — CHANGES_REQUESTED** [review]

File: src/app/switcher/switcher.service.ts (hunk 4 above)
Comment (trtshen) [reply src/app/switcher/switcher.service.ts]: @sasangachathumal  https://github.com/intersective/app-v2/tree/hotfix/CORE-4694/primary-2ndary-colors

Comment (trtshen) [reply src/app/switcher/switcher.service.ts]: @sasangachathumal i've created a branch  to make your work easier, it has all the color changes code i did. you can merge this changes back into your current work later.

Comment (sasangachathumal) [reply src/app/switcher/switcher.service.ts]: sure. thank you @trtshen .

File: src/app/shared/request/request.service.ts [hunk 6]
```diff
@@ -150,29 +158,31 @@ export class RequestService {
       );
   }
 
-  post(endPoint: string = '', data, httpOptions?: RequestOptions, isLoginAPI?: boolean): Observable<any> {
-    if (!httpOptions) {
-      httpOptions = {};
+  post(params: POSTParams): Observable<any> {
```
Comment (trtshen) [inline src/app/shared/request/request.service.ts:161]: @sasangachathumal  👍  i like how this got cleaned up!

Comment (sasangachathumal) [reply src/app/shared/request/request.service.ts:161]: Thank you @trtshen 👍. Need to do same refactoring for other functions like get in the future.

**Pull request merged**